### PR TITLE
Erin/actionitem mailer

### PR DESCRIPTION
**Pull request by erinysong**
## Email functionality when assignments are created

After a staff creates an assignment, an email is sent to the student associated with the assignment. Currently the emailer doesn't work because there are some issues with the action item controller not saving assignments when they're created from the admin dashboard, but we are currently testing if the mailer at least works from the front end
### Related PRs

Optional - any related PRs you're waiting on, or PRs that will conflict, etc

### Migrations

Optional - if you added anything to the database through migration(s)

### Screenshots

Optional but would be nice for final presentation :)

CC: @didvi

### CODE REVIEW

File: app/controllers/api/assignments_controller.rb [hunk 1]
```diff
@@ -77,6 +78,7 @@ def create_template
 
         if @template.save
             render json: @template, status: :created
+
```
Comment (wilsonw926) [inline app/controllers/api/assignments_controller.rb:81]: rm this extra line

File: app/mailers/assignment_mailer.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class AssignmentMailer < ApplicationMailer
+    
+    def new_assignment
+        @assignment = params[:assignment]
+        @url = ENV['MAIL_HOST']
```
Comment (wilsonw926) [inline app/mailers/assignment_mailer.rb:5]: I think we need to differentiate between development MAIL_HOST (localhost:3000) and production (the heroku link). Should we create two env variables.

File: app/mailers/assignment_mailer.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+        mail(to: "unlooptestparticipant@gmail.com", subject: '[Unloop] New Action Item Assigned')
```
Comment (wilsonw926) [inline app/mailers/assignment_mailer.rb:6]: still hardcoded :/

File: app/mailers/assignment_mailer.rb (hunk 2 above)
Comment (erinysong) [reply app/mailers/assignment_mailer.rb:5]: ooh all right, would both the env variables refer to localhost:3000 and just be under different names then?

Comment (wilsonw926) [reply app/mailers/assignment_mailer.rb:5]: Mm I was thinking we would want a dev_mail_host that is local host 3000 and a prod_mail_host car that is https://unloop-staging.herokuapp.com/.

Comment (erinysong) [reply app/mailers/assignment_mailer.rb:5]: got it, I just included a prod_mail_host to link to the heroku, do you have any suggestions on which one to use for the url in the assignment mailer? or do we have to use both

Comment (wilsonw926) [reply app/mailers/assignment_mailer.rb:5]: Yeah you can do a check 

```Rails.env.production?```

**Comment by erinysong** [comment]
instructions for testing mailer on local environment:
https://mailcatcher.me/

**Comment by erinysong** [comment]
a test component for this mailer is included in erin/actionitem-mailer-tests. in order to test the mailer, do the following in the actionitem-mailer-tests branch:

1. in terminal: `gem install mailcatcher`
2. in terminal: `mailcatcher`
3. Go to http://localhost:1080/
4. In a new tab, go to localhost:3000/assignments. Click on the create button. This should send an api call to assignment create
5. In the mailer tab (localhost:1080), there should be email logs for the users the assignments are assigned to

More references at https://mailcatcher.me/

**Pull request merged**